src/main/java/hudson/plugins/sshslaves/verifiers/ManuallyProvidedKeyVerificationStrategy.java

```diff
@@ -92,18 +93,36 @@ private static HostKey parseKey(String key) {
                 RSASHA1Verify.decodeSSHRSAPublicKey(keyValue);
             } else if ("ssh-dss".equals(algorithm)) {
                 DSASHA1Verify.decodeSSHDSAPublicKey(keyValue);
+            } else if (isClass("com.trilead.ssh2.signature.ED25519KeyAlgorithm") && "ssh-ed25519".equals(algorithm)) {
+                decodePublicKey("com.trilead.ssh2.signature.ED25519KeyAlgorithm", keyValue);
+            } else if (isClass("com.trilead.ssh2.signature.ECDSAKeyAlgorithm") && "ecdsa-sha2-nistp256".equals(algorithm)) {
+                decodePublicKey("com.trilead.ssh2.signature.ECDSAKeyAlgorithm", keyValue);
+            } else if (isClass("com.trilead.ssh2.signature.ECDSAKeyAlgorithm") && "ecdsa-sha2-nistp384".equals(algorithm)) {
+                decodePublicKey("com.trilead.ssh2.signature.ECDSAKeyAlgorithm", keyValue);
+            } else if (isClass("ECDSAKeyAlgorithm") && "ecdsa-sha2-nistp521".equals(algorithm)) {
+                decodePublicKey("com.trilead.ssh2.signature.ECDSAKeyAlgorithm", keyValue);
             } else {
-                throw new IllegalArgumentException("Key algorithm should be one of ssh-rsa or ssh-dss");
+                throw new IllegalArgumentException(Messages.ManualKeyProvidedHostKeyVerifier_UnknownKeyAlgorithm());
             }
-        } catch (IOException ex) {
-            throw new IllegalArgumentException(Messages.ManualKeyProvidedHostKeyVerifier_KeyValueDoesNotParse(algorithm), ex);
-        }  catch (StringIndexOutOfBoundsException ex) {
-            // can happen in DSASHA1Verifier with certain values (from quick testing)
+        } catch (IOException | StringIndexOutOfBoundsException | NoSuchMethodException | IllegalAccessException | InvocationTargetException | ClassNotFoundException ex) {
             throw new IllegalArgumentException(Messages.ManualKeyProvidedHostKeyVerifier_KeyValueDoesNotParse(algorithm), ex);
         }
         
         return new HostKey(algorithm, keyValue);
     }
+
+    private static void decodePublicKey(String className, byte[] keyValue) throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        Class.forName(className).getMethod("decodePublicKey", byte[].class).invoke(keyValue);
```

Comment:
Some logging on the FINE/FINEST level would be really useful. Just to know when the new algorithm decoding fails